Added the remaining requested bash scripts

rleenay opened the pull request:
Hello! Thank you for making a change to `packer-images`. To make sure all the
instances don't clash with each other

- [ ] changed "ami_name" to something descriptive and without spaces
- [ ] changed "ami_description" to something descriptive
- [ ] `packer build image.json` works
- [ ] Instance runs

jameshowardwang commented:
Why the custom $HOME/bin and Ubuntu/programs dirs instead of usr/local/bin or /usr/bin? Do none of these programs have “make install” options?

rleenay commented:
Quite a few of these programs are pre-installed binaries already. 

JamesWebber and I discussed that it would be good to just make a single $HOME/bin folder where they can all be easily identified. We could definitely push them to usr/local/bin or /usr/bin instead if there would be a need to get them out of the home directory

jamestwebber commented:
After some reflection `/usr/local/bin` is probably the best choice for all of this stuff. It's the standard place for users to install software, and it doesn't require creating a directory and adding it to the path.

rleenay commented:
Sounds great James, let me update all of my pull requests with that new destination and we can move on from there.

jamestwebber commented:
All right. Just FYI you probably need to do `sudo cp` or `sudo mv` to get stuff in there.

rleenay commented:
James, I changed all of the directories to /usr/local/bin. I also added pip to the necessary script. 

I'll let you make the same change to bowtie, and then we can approve all of them and push it through.

jamestwebber commented:
As I mentioned in #15, I'd prefer to standardize python usage, either with anaconda3 or miniconda3 (probably the former&mdash;it brings in a lot of stuff but we use a lot of it). So the script to install biopython should use `conda install -c bioconda biopython`.

And of course it needs to be installed in the presence of anaconda3. There's already an anaconda image in the repo but I'm going to up it to 5.0.1 and give it a standard name (rather than based on `user`). I don't know if installing biopython even needs its own script at that point.

rleenay commented:
No problem, I didn't understand that the first time around, thank you for explaining. 

For biopython, let's just not install it then. It should install pretty quickly if someone needs it. Unfortunately, I'm not sure how to just remove that script, as I combined all of these in the same pull request? Should I just go in and delete the text in the script?

jamestwebber commented:
On the command line you can use `git rm biopython.sh` to remove the file, and then commit and push the change to the `rleenay-patch-4` branch.

Or from GitHub.com you can [view the file](https://github.com/czbiohub/packer-images/blob/rleenay-patch-4/scripts/biopython.sh) and there's a little 'delete this file' icon at the right.

rleenay commented:
That should be the long and short of it then. There are only two remaining items:

1.) Carrying over S3 authentication to the booted image
2.) Determining how we want to handle reference sequences. As scripts to run at the beginning of the code, knowing that you'll need to allow time to download? Or include them with packer.
